Onboarding refactor

nilsonsasaki opened the pull request:
Used pr #73 to lessen TextStyle uses

Removed circle page indicator from third page
updated Button on third page to an AppButton

Added a new forth page to redirect to "set user name"

Added const where it was possible

Now, you can navigate to "set user name" swiping left on third page or by the Button

(set user name will need an exit game popup to avoid returning to an empty screen)

yuumasato commented on lib/ui/onboarding.dart:
```diff
@@ -32,102 +34,111 @@ class _OnboardingScreenState extends State<OnboardingScreen> {
+                    child: SvgPicture.asset('assets/images/logo.svg',
+                        width: mediaWidth / 3))),
+            ),
+            Positioned(
+                left: 0.0,
+                right: 0.0,
+                bottom:  80.0,
+                child: CirclePageIndicator(
+                  size: 10.0,
+                  selectedSize: 12.0,
+                  itemCount: 3,
+                  currentPageNotifier: _currentPageNotifier,
+                )),
```
The CirclePageIndicator is being swiped together with the page, it is supposed to stay still.

yuumasato commented on lib/ui/onboarding.dart:
```diff
@@ -32,102 +34,111 @@ class _OnboardingScreenState extends State<OnboardingScreen> {
+                  child: SizedBox(
+                      width: mediaWidth / 3,
+                      height: 48,
+                      child: AppButton(
+                          onPressed: () {
+                            Get.toNamed(UserNameScreen.route);
+                          },
+                          text: 'continue'.tr,
```
The "Continue" button can be removed.

yuumasato reviewed (changes requested):
> Removed circle page indicator from third page

As we are continuing the swiping action, the page action should probably show 4 dots, instead of 3.

> updated Button on third page to an AppButton
The designers recommended the button be removed, :)

> (set user name will need an exit game popup to avoid returning to an empty screen)

Maybe the user should be allowed to swipe back and go back to the onboarding?
If that is not easy, it should be fine for the user to be "locked" in the "SetUserNameScreen".

nilsonsasaki commented:
moved back CiclePage indicator to a stack on top of page view 
added a forth dot to Circle Page Indicator
changed empty scaffold to a UserNameScreen (used set_user_name as a widget maybe cause performance issues)

removed AppButton
